### ISSUE TITLE
Ignore invalid return when jsdoc @abstract is present

### ISSUE THREAD

**Issue by SimonSchick** [eslint/eslint#5941]
**What version of ESLint are you using?**
2.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**What did you do? Please include the actual source code causing the issue.**

I have somewhat of an abstract class where I have a method such as this:

```javascript
    /**
     * Creates a new OAuth grant using the provided query and permissions. At
     * this point the query and permissions will already have been validated.
     * @abstract
     * @param  {Object} query
     * @param  {String[]} permissions
     * @return {Promise.<{code: String, state: String}>}
     */
    createOAuth (query, permissions) {
        throw new Error('not implemented');
    }
```

**What did you expect to happen?**

Ideally with JSDoc enabled, eslint should ignore invalid returns when `@abstract` is set.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint complains about a the missing return in the function body eg. `Unexpected @return tag`.

**Comment by nzakas** [comment]
Good point, I think this is a good idea. Basically any method marked as `@abstract` should not check the return statement. 

@SimonSchick can you submit a PR for this?

**Comment by SimonSchick** [comment]
Could you tell me how I can only run a part of the test suite? I tried to run `npm test -- -f "valid-jsdoc"` which is the default way in mocha.

**Comment by kaicataldo** [comment]
@SimonSchick Thanks for working on this!

I usually use this command: `./node_modules/.bin/mocha tests/lib/path/to/tests.js`

Source: http://eslint.org/docs/developer-guide/unit-tests#running-individual-tests
